### ISSUE TITLE
(Re) Implementing path-value autocompletion

### ISSUE THREAD

**Issue by ddeltree** [ddeltree/leavify#9]
Forgot to mention in the commit message, but ba1cd1f deletes the functionality which autocompletes a leaf-path and its leaf-value.
The reason for it is that typescript gets too slow. "To much type instantiation", thinking it was possibly infinite.

I think the problem was the use of three recursive types in a row, namely `Fragment<T> = Tree<RecursivePartial<T>>`, and finally `LeafPath<Fragment<T>>`. The solution was to just ask whether `T` extends `Fragment`.

So... I'll be getting back to this later, and check whether the lagging continues. Specifically, the deleted types are:
1. `LeafValue` - which autocompletes for the possible values (when using `as const`) or their general type.
2. `PathLeafPair` - the path and either its `as const` value or its general type.

```ts
const sample = [42, {prop: 'value'}] as const;
const pair: PathLeafPair<typeof sample> = ['[1].prop', 'value'] // or ['[0]', 42]
```

**Comment by davi-alexandre** [comment]
> The reason for it is that typescript gets too slow. "To much type instantiation", thinking it was possibly infinite.

The problem persists, specifically for the `any` generic argument, even though the resultant type is always `any`.

**Comment by ddeltree** [comment]
resolved 3a3d5f1d20a066c43c632a4e34e8cc1c13257a46
